### PR TITLE
Integrate GolangCI-lint

### LINKED ISSUE #21: Integrate GolangCI-Lint
Mattermost is currently in the process of using [`golangci-lint`](https://github.com/golangci/golangci-lint/) as the default linter for all go projects. https://github.com/mattermost/mattermost-plugin-autolink/pull/108 serves as an example of how the migration should look like:
1. Copy `.golangci.yml` from https://github.com/mattermost/mattermost-plugin-autolink/ into this repository.
2. Copy `Makefile` from https://github.com/mattermost/mattermost-plugin-autolink/ into this repository.
3. Copy `build/` from https://github.com/mattermost/mattermost-plugin-autolink/ into this repository.
4. Run `go mod tidy` and `make dist` to ensure the build system works fine.
5. Update `linters-settings.goimports.local-prefixes` to the go module path i.e. to `github.com/mattermost/mattermost-plugin-aws-SNS`.
6. Run `make golangci-lint` and fix all issues that arise.

Fixing an linter issue is sometimes trivial but can also be quite complicated. There might be cases, where ignoring an issue using [`issues.exclude-rules`](https://github.com/golangci/golangci-lint/#config-file) is needed. If you don't know how to resolve an error, please reach out to `@hanzei` on the [Mattermost Community Server](https://community.mattermost.com) or ask the question here.

### PR DESCRIPTION
**Pull request by nathanaelhoun**
#### Summary

This PR copies most of the golangci-lint configuration from mattermost/mattermost-plugin-starter-template#90 into this repo. This helps with a consistent code style and should improve code quality over time.

Only one change was needed : permission for manifest file. See https://github.com/mattermost/mattermost-plugin-starter-template/issues/95.

Inspired by mattermost/mattermost-plugin-autolink#108.

#### Ticket Link

Closes #21

### CODE REVIEW

**Review by iomodo — APPROVED** [review]
LGMT @nathanaelhoun ! thanks for the contribution.

File: build/custom.mk [hunk 1]
```diff
@@ -1 +1 @@
-# Include custom targets and environment variables here
+# Include custome targets and environment variables here
```
Comment (nathanaelhoun) [inline build/custom.mk:1]: I need to fix this typo

**Review by hanzei — APPROVED** [review]
:+1:

**Comment by hanzei** [comment]
Skipping QA review as there are no changes to the product.

**Pull request merged**